File: iOS/LiveStreamCore/Sources/State/CoHostState.swift
//
//  CoHostState.swift
//  LiveStreamCore
//
//  Created by jeremiawang on 2024/10/22.
//

import RTCRoomEngine

public struct CoHostState: State {
    public var connectedUserList: [TUIConnectionUser] = []
    public var sentConnectionRequestList: [TUIConnectionUser] = []
    public var receivedConnectionRequest: TUIConnectionUser? = nil
    
    public var enableConnection: Bool = true
    
    public init() {}
}

File: Android/tuilivekit/livestreamcore/src/main/java/com/trtc/uikit/livekit/livestreamcore/state/CoHostState.java
package com.trtc.uikit.livekit.livestreamcore.state;

import com.tencent.cloud.tuikit.engine.extension.TUILiveConnectionManager.ConnectionUser;
import com.trtc.tuikit.common.livedata.LiveData;

import java.util.ArrayList;
import java.util.List;

public class CoHostState {
    public final LiveData<List<ConnectionUser>> connectedUserList         = new LiveData<>(new ArrayList<>());
    public final LiveData<List<ConnectionUser>> sentConnectionRequestList = new LiveData<>(new ArrayList<>());
    public final LiveData<ConnectionUser>       receivedConnectionRequest = new LiveData<>();
    public       boolean                        enableConnection          = true;

    public void reset() {
        connectedUserList.get().clear();
        sentConnectionRequestList.get().clear();
        receivedConnectionRequest.set(null);
    }
}
